src/connection/host_lookup.c

```diff
@@ -0,0 +1,670 @@
+	  fprintf (stdout, "%s\n", er_msg ());
+	}
+      return NULL;
+    }
+
+  return hp;
+}
+
+static int
+load_hosts_file ()
+{
+  FILE *fp;
+  char file_line[LINE_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token, temp_token[LINE_BUF_SIZE + 1];
+  char *save_ptr_strtok;
+  /*delimiter */
+  const char *delim = " \t\n";
+  char ipaddr[IPADDR_LEN];
+  char hostname[HOSTNAME_LEN];
+  int cache_idx = 0, temp_idx;
+
+  char addr_trans_ch_buf[IPADDR_LEN];
+  struct in_addr addr_trans;
+
+  /*True, when the string token has hostname, otherwise, string token has IP address */
+  bool hostent_flag;
+
+  memset (file_line, 0, LINE_BUF_SIZE);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, user_hosts_File);
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_MOUNT_FAIL, 1, host_conf_file_full_path);
+      fprintf (stdout, "%s\n", er_msg ());
+      return LOAD_FAIL;
+    }
+
+  while (fgets (file_line, LINE_BUF_SIZE, fp) != NULL)
+    {
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      hostent_flag = INSERT_IPADDR;
+
+      do
+	{
+	  if (token == NULL || *token == '#')
+	    {
+	      break;
+	    }
+	  if (hostent_flag == INSERT_IPADDR)
+	    {
+	      strcpy (temp_token, token);
+	      if (ip_format_check (temp_token) == false)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_IP_ADDR_INVALID_FORMAT, 1, token);
+		  fprintf (stdout, "%s\n", er_msg ());
+
+		  user_host_Map.clear ();
+
+		  fclose (fp);
+
+		  return LOAD_FAIL;
+		}
+	      strncpy (ipaddr, token, IPADDR_LEN);
+	      ipaddr[IPADDR_LEN - 1] = '\0';
+
+	      hostent_flag = INSERT_HOSTNAME;
+	    }
+	  else
+	    {
+	      if (strlen (token) > HOSTNAME_LEN - 1)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_HOST_NAME_TOO_LONG, 1, token);
+		  fprintf (stdout, "%s\n", er_msg ());
+
+		  user_host_Map.clear ();
+
+		  fclose (fp);
+
+		  return LOAD_FAIL;
+		}
+	      strcpy (hostname, token);
+	    }
+	}
+      while ((token = strtok_r (NULL, delim, &save_ptr_strtok)) != NULL);
+
+      if (strcmp ("\0", hostname) && strcmp ("\0", ipaddr))
+	{
+	  /*not duplicated hostname */
+	  if ((user_host_Map.find (hostname) == user_host_Map.end ()))
+	    {
+	      user_host_Map[hostname] = cache_idx;
+	      user_host_Map[ipaddr] = cache_idx;
+	      hostent_Cache[cache_idx] = hostent_alloc (ipaddr, hostname);
+
+	      cache_idx++;
+	    }
+	  /*duplicated hostname */
+	  else
+	    {
+	      /*duplicated hostname but different ip address */
+
+	      temp_idx = user_host_Map.find (hostname)->second;
+	      memcpy (&addr_trans.s_addr, hostent_Cache[temp_idx]->h_addr_list[0], sizeof (addr_trans.s_addr));
+
+	      if (inet_ntop (AF_INET, &addr_trans, addr_trans_ch_buf, sizeof (addr_trans_ch_buf)) == NULL)
```

Comment:
not **addr_trans**, but **addr_trans.s_addr**

`if (inet_ntop (AF_INET, &addr_trans.s_addr, addr_trans_ch_buf, sizeof (addr_trans_ch_buf)) == NULL)`

Reply:
Right. Previous one worked but it is not adequate variable.  Thank you. I will apply it.